packages/compass-data-modeling/src/components/export-diagram-modal.tsx

```diff
@@ -64,25 +66,68 @@ const ExportDiagramModal = ({
   const [exportFormat, setExportFormat] = useState<'png' | 'json' | null>(null);
   const diagram = useDiagram();
   const [isExporting, setIsExporting] = useState(false);
+  const abortControllerRef = useRef<AbortController | null>(null);
+  const toast = useToast();
+  useEffect(() => {
+    const cleanup = () => {
+      if (abortControllerRef.current) {
+        abortControllerRef.current.abort();
+        abortControllerRef.current = null;
+      }
+    };
+    const abortController = new AbortController();
+    if (isModalOpen) {
+      abortControllerRef.current = abortController;
+    } else {
+      cleanup();
+    }
+    return cleanup;
+  }, [isModalOpen]);
+
+  const onClose = useCallback(() => {
+    setExportFormat(null);
```

Comment:
How do you feel about remove this line? Keeping around the last format could be nice for users that are repeating the same action for different diagrams. Less code that way too. Not a blocker by any means, fine as is.

Reply:
No strong preferences here. Happy to preserve this state